### ISSUE TITLE
Add a follow-up overview to the event participants screen of individual events [2]

### ISSUE THREAD

**Issue by markusmann-vg** [SORMAS-Foundation/SORMAS-Project#4492]
### Situation Description

After #3003 there will be a feature to document follow ups for event participants.

### Feature Description
As we have the follow-up view for Cases/contacts we would need it for events as well.

**Comment by MateStrysewske** [comment]
* Add a switch to the event participants view inside an event (similar to the switch that allows to toggle between default/detailed/follow-up in case directory) with the options "Event participants" and "Follow-up"; "Event participants" is the default and is exactly what's displayed right now, "Follow-up" adds an alternative directory; put the switch to the left of the bulk actions dropdown
* Implement a follow-up grid view similar to the one we already have in the case/contact directories and that shows the follow-up overview of every event participant in the event; should include all elements that are also part of the already existing follow-up overviews (+ all filters of the event participants view)
* Use the same columns as in the case follow-up overview except the symptom journal status

**Comment by bernardsilenou** [comment]
@MateStrysewske @kwa20 
I do not know if i got you wall,  do you mean  we should add the followup view to the individual event?
In practice, we followup up the ep person and do not know the events that they are linked to in advance. For contacts, we do not have a followup view of contact within a case.
- I think we should replace the first point with an event participant directory and maybe place it beside the action icon of the event directory.

**Comment by MateStrysewske** [comment]
@bernardsilenou If you think that a follow-up overview of individual events isn't helpful, then I agree we should rather start to implement an event participant directory and add the follow-up overview to that. @kwa20 Do you agree with Bernard?

**Comment by bernardsilenou** [comment]
@kwa20 @MateStrysewske It is a good thing to have in future. Based on priority at the moment, I would suggest we start with ep directory + followup  of all participants just as we do for contacts.

**Comment by kwa20** [comment]
@bernardsilenou @MateStrysewske Since we have a person directory now, we could actually think about making follow-up purely person based and transfer the whole follow-up view to persons. This might be a big change but worth thinking about because it might also take care of following-up persons rather then entities where one person can have many.

**Comment by MateStrysewske** [comment]
@kwa20 Follow-up is person-based already. If a person has two contacts with the same disease in a similar time frame, created visits will appear in both contacts. Event participant follow-up would follow the same logic. Cases, contacts and event participants definitely still need their individual follow-up views because certain user roles will primarily work with these directories, not with the person directory.

**Comment by kwa20** [comment]
@MateStrysewske Yes, the adoption of follow-up visits is not the problem that's right. I was thinking about something along the lines of displaying the summarized follow-up period depending on the person because the current implementation is per contact:

<img width="847" alt="personBasedFollowUp" src="https://user-images.githubusercontent.com/70317594/114607682-1d482d00-9c9d-11eb-8669-6b18dc57a85d.PNG">

That makes it quite hard to keep an overview on the exact follow-up duration of the person if it's not deliberately aligned like this on the follow-up view. Furthermore, it's hard for people to follow-up on e.g. households with cases and contacts under follow-up because it's not possible to display it across entities. We could ensure individual follow-up views with toggle options like seen for persons:

<img width="226" alt="personBasedFollowUp2" src="https://user-images.githubusercontent.com/70317594/114608002-7dd76a00-9c9d-11eb-8310-eda51fa55084.PNG">
 
I was planning to create another issue for this anyway but it would affect this one as well.

**Comment by bernardsilenou** [comment]
@kwa20 
- You are very correct and I  think the person view should even be the default followup view for cases, contacts, ep, etc
- We can add a toggle to display either in person or entity view. For person view, we select unique person uuid and colour cells based on all  the followups of the person. If followups overlap, as in diagram above, then we colour only once, if not we colour the beginning and end of each interval but still maintain all on one row. the contact uuid can then be replaced by person uuid in the person view.
